### PR TITLE
nixosTests.chromium: Port to Python

### PR DESCRIPTION
**Pull request by tfc**
###### Motivation for this change

#72828

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tfc** [comment]
@GrahamcOfBorg test chromium

File: nixos/tests/chromium.nix [hunk 1]
```diff
@@ -47,155 +47,218 @@ mapAttrs (channel: chromiumPkg: makeTest rec {
...
     in "${pkgs.xdotool}/bin/xdotool '${xdoScript}'";
   in ''
+    import shlex
+    from contextlib import contextmanager, _GeneratorContextManager
+
+
     # Run as user alice
-    sub ru ($) {
-      my $esc = $_[0] =~ s/'/'\\${"'"}'/gr;
-      return "su - alice -c '$esc'";
```
Comment (worldofpeace) [inline nixos/tests/chromium.nix]: can we not hardcode the `alice` user name into the test?
If we ever change it's going to be a pain, we can use 
https://github.com/NixOS/nixpkgs/blob/1013c2e3a6f4a94305047999537c45d8853572de/nixos/tests/gnome3.nix#L29 `name`.

**Comment by tfc** [comment]
@GrahamcOfBorg test chromium

**Comment by flokli** [comment]
x86_64 timed out as it tried to build chromium. As aarch64 succeeded, LGTM.

**Pull request merged**